### PR TITLE
Add state sampler.

### PR DESCRIPTION
**Pull request by sergeisakov**

### CODE REVIEW

File: tests/statespace_basic_test.cc [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+#include "gtest/gtest.h"
+
+#include "../lib/parfor.h"
+#include "../lib/simulator_basic.h"
+#include "../lib/statespace_basic.h"
+
+namespace qsim {
+
+TEST(StateSpaceAVXTest, SamplingBasic) {
```
Comment (95-martin-orion) [inline tests/statespace_basic_test.cc]: `TEST(StateSpaceBasicTest, ...)` instead? Same below.

File: tests/BUILD [hunk 2]
```diff
@@ -88,10 +88,30 @@ cc_test(
     ],
 )
 
+cc_library(
```
Comment (95-martin-orion) [inline tests/BUILD]: I would recommend adding `testonly = 1,` to this rule (see [bazel docs](https://docs.bazel.build/versions/master/be/common-definitions.html) for testonly usage).

File: tests/statespace_test.h [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+#include "../lib/circuit_reader.h"
+#include "../lib/fuser_basic.h"
+#include "../lib/gates_def.h"
+#include "../lib/io.h"
+#include "../lib/run_qsim.h"
+
+namespace qsim {
+
+template <typename StateSpace>
+void TestSamplingBasic() {
```
Comment (95-martin-orion) [inline tests/statespace_test.h]: Minor nit: `TestSamplingBasic` suggests to me that this is for `StateSpaceBasic` only - maybe `TestSamplingSmall`?

**Comment by sergeisakov** [comment]
Done. Thanks for your review!

**Pull request merged**